doc/source/tutorial/interpolate.rst

```diff
@@ -10,38 +10,494 @@ Interpolation (:mod:`scipy.interpolate`)
+the interpolation of a :ref:`derivative of a sine function<tutorial-interpolate_ppoly>`
+with b-splines:
+
+.. plot::
+
+    >>> x = np.linspace(0, 3/2, 7)
+    >>> y = np.sin(np.pi*x)
+
+    To construct the interpolating objects given data arrays, ``x`` and ``y``,
+    we use the `make_interp_spline` function:
+
+    >>> from scipy.interpolate import make_interp_spline
+    >>> bspl = make_interp_spline(x, y)
+
+    Now ``bspl`` is a `BSpline` object which has an interface similar to `PPoly`.
+    In particular, it can be evaluated at a data point and differentiated:
+
+    >>> der = bspl.derivative()      # a BSpline representing the derivative
+    >>> import matplotlib.pyplot as plt
+    >>> xx = np.linspace(0, 3/2, 51)
+    >>> plt.plot(xx, bspl(xx), '--', label=r'$\sin(\pi x)$ approx')
+    >>> plt.plot(x, y, 'o', label='data')
+    >>> plt.plot(xx, der(xx)/np.pi, '--', label='$d \sin(\pi x)/dx / \pi$ approx')
+    >>> plt.legend()
+    >>> plt.show()
+
+Note that by default `make_interp_spline` constructs a cubic spline;
+its derivative is a quadratic:
+
+    >>> bspl.k, der.k
+    (3, 2)
+
+This way, the default result of ``make_interp_spline(x, y)`` is equivalent to
+``CubicSpline(x, y)``. The difference is that the former allows several optional
+capabilities: it can construct splines of various degrees (via the optional
+argument ``k``) and predefined knots (via the optional argument ``t``). 
+
+A b-spline of degree ``k`` is defined by its knots and coefficients. The knots
+are availble as the ``t`` attribute of a `BSpline` instance:
+
+    >>> print(bspl.t)
+    [0.  0.  0.  0.        0.5  0.75  1.        1.5  1.5  1.5  1.5 ]
+    >>> print(x)
+    [            0.  0.25  0.5  0.75  1.  1.25  1.5 ]
+
+We see that the knot vector by default is constructed from the input
+array ``x``: first, it is made :math:`(k+1)` -regular (it has ``k+1``
+repeated knots appended and prepended); then, the second and
+second-to-last points of the input array are removed---this is the so-called
+*not-a-knot* boundary condition. 
+
+In general, an interpolating spline of degree ``k`` needs
+``len(t) - len(x) - k - 1`` boundary conditions. For cubic splines with
+``(k+1)``-regular knot arrays this means two boundary conditions---or
+removing two values from the ``x`` array. Various boundary conditions can be
+requested using the optional ``bc_type`` argument of `make_interp_spline`.
+
+The b-spline coefficients are accessed via the ``c`` attribute of a `BSpline`
+object:
+
+    >>> len(bspl.c)
+    7
+
+The convention is that for ``len(t)`` knots there are ``len(t) - k - 1``
+coefficients. Some routines (see the :ref:`Smoothing splines section
+<tutorial-interpolate_fitpack>`) zero-pad the ``c`` arrays so that
+``len(c) == len(t)``. These additional coefficients are ignored for evaluation.
+
+Also note that the coeffients are given in the
+:ref:`b-spline basis <tutorial-interpolate_bspl_basis>`, not the power basis
+of :math:`1, x, \cdots, x^k`.
```

Comment:
I got a bit lost in here. Is it assumed that the reader knows what a "knot" is? 
There seems to be a lot of detail here about how the b-splines are working internally without a punchline (how the user can use it to do something).

Reply:
Sigh. Need to think how to restructure this whole section. Tried to avoid going into a lengthy teaching session, and wiggle my way through at least two sets of readers (those who see it for the first time and those who look for details), trying to document our spline format conventions (there are several in the literature and details need to be fished out, which is frustrating) and giving an overview of the functionality. Clearly, this was not a success :-).

Reply:
> Tried to avoid going into a lengthy teaching session, and wiggle my way through at least two sets of readers 

Yeah, I'm the kind who sees it for the first time.

I'm not sure how to do both at once. I feel like the API reference is for those who know the theory, and tutorial is for those who don't.

Reply:
> Yeah, I'm the kind who sees it for the first time.

Then you are a perfect reader of this! Seriously.

Mixing up `np.Polynomial` here would be a bit confusing I suspect. Note that docstrings of both `PPoly` and `BSpline` list pure-python illustrations. I tried to refer to the docstrings, let me know if it worked. Or would it help to copy these implementations to the tutorial?